### ISSUE TITLE
Latest releases (1.23.x, 1.24.x) missing from NuGet. Invalid 2.x versions show up

### ISSUE THREAD

**Issue by clehene** [grpc/grpc#20563]
None of these can be found in NuGet

**Comment by clehene** [comment]
Also there are 2.x versions that seem to overlap wit grpc-dotnet versions. 
This is highly confusing. Dependency updates are suggested for 2.x releases and after upgrade we're seeing 
```
 ---> System.TypeInitializationException: The type initializer for 'Google.Api.AnnotationsReflection' threw an exception.
 ---> System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at Google.Protobuf.Reflection.ExtensionCollection.<>c__DisplayClass2_0.<.ctor>b__0(FieldDescriptorProto extension, Int32 i)
   at Google.Protobuf.Reflection.DescriptorUtil.ConvertAndMakeReadOnly[TInput,TOutput](IList`1 input, IndexedConverter`2 converter)
   at Google.Protobuf.Reflection.ExtensionCollection..ctor(FileDescriptor file, Extension[] extensions)
   at Google.Protobuf.Reflection.FileDescriptor..ctor(ByteString descriptorData, FileDescriptorProto proto, IEnumerable`1 dependencies, DescriptorPool pool, Boolean allowUnknownDependencies, GeneratedClrTypeInfo generatedCodeInfo)
   at Google.Protobuf.Reflection.FileDescriptor.BuildFrom(ByteString descriptorData, FileDescriptorProto proto, FileDescriptor[] dependencies, Boolean allowUnknownDependencies, GeneratedClrTypeInfo generatedCodeInfo)
   at Google.Protobuf.Reflection.FileDescriptor.FromGeneratedCode(Byte[] descriptorData, FileDescriptor[] dependencies, GeneratedClrTypeInfo generatedCodeInfo)
   at Google.Api.AnnotationsReflection..cctor()
   --- End of inner exception stack trace ---
```

**Comment by clehene** [comment]
![image](https://user-images.githubusercontent.com/101597/66596663-29a47480-eb52-11e9-91fa-685b6bace5af.png)

**Comment by clehene** [comment]
![image](https://user-images.githubusercontent.com/101597/66597039-f9110a80-eb52-11e9-923d-68d48f30dedf.png)

**Comment by clehene** [comment]
> Also there are 2.x versions that seem to overlap wit grpc-dotnet versions.
> This is highly confusing. Dependency updates are suggested for 2.x releases and after upgrade we're seeing
> 
> ```
>  ---> System.TypeInitializationException: The type initializer for 'Google.Api.AnnotationsReflection' threw an exception.
>  ---> System.IndexOutOfRangeException: Index was outside the bounds of the array.
>    at Google.Protobuf.Reflection.ExtensionCollection.<>c__DisplayClass2_0.<.ctor>b__0(FieldDescriptorProto extension, Int32 i)
>    at Google.Protobuf.Reflection.DescriptorUtil.ConvertAndMakeReadOnly[TInput,TOutput](IList`1 input, IndexedConverter`2 converter)
>    at Google.Protobuf.Reflection.ExtensionCollection..ctor(FileDescriptor file, Extension[] extensions)
>    at Google.Protobuf.Reflection.FileDescriptor..ctor(ByteString descriptorData, FileDescriptorProto proto, IEnumerable`1 dependencies, DescriptorPool pool, Boolean allowUnknownDependencies, GeneratedClrTypeInfo generatedCodeInfo)
>    at Google.Protobuf.Reflection.FileDescriptor.BuildFrom(ByteString descriptorData, FileDescriptorProto proto, FileDescriptor[] dependencies, Boolean allowUnknownDependencies, GeneratedClrTypeInfo generatedCodeInfo)
>    at Google.Protobuf.Reflection.FileDescriptor.FromGeneratedCode(Byte[] descriptorData, FileDescriptor[] dependencies, GeneratedClrTypeInfo generatedCodeInfo)
>    at Google.Api.AnnotationsReflection..cctor()
>    --- End of inner exception stack trace ---
> ```

Is caused by https://github.com/protocolbuffers/protobuf/issues/6749

**Comment by jtattermusch** [comment]
Starting from "1.23", gRPC C# made a major version bump and became v2.23.0
The background and the rationale for that decision is described here: https://github.com/grpc/proposal/blob/master/L57-csharp-new-major-version.md

**Comment by clehene** [comment]
@jtattermusch ok, but why close when the last two releases are still missing from NuGet?

**Comment by jtattermusch** [comment]
They are not missing. v2.23 and v2.24 are the C# nugets that correspond to the 1.23 and 1.24 branches in this repository. There are several languages that live in this repository and we only have one release process (we release packages for all languages in this repository at the same time) so C# 2.23 is actually part of the 1.23.x release train.

Additional context is here:
https://groups.google.com/forum/#!msg/grpc-io/tCTlDjG_Vsg/NlRZVMMqAQAJ

**Comment by clehene** [comment]
@jtattermusch perhaps I'm missing something, then.

The additional context mentions:

> The release process and the release cadence for gRPC C# stays unchanged (which means that 2.23.x will be released at the same time as the 1.23.x version of other languages, gRPC C# 2.24.x will be released along with 1.24.x etc.)

In grpc i see:

* 1.24.0
* 1.24.1 
* 1.24.2

Can you point to the corresponding 2.24.x releases in NuGet?

**Comment by jtattermusch** [comment]
Ah, I didn't realize you were asking about the 2.24.1 and 2.24.2 patch releases specifically - these hasn't been published for C# yet. I'll check if those patches have anything that's relevant to C# (it might not) and push the nugets if needed. Until then you can use 2.24.0 version from nuget (which does exist https://www.nuget.org/packages/Grpc.Core/2.24.0)
